gobblin-metrics-libs/gobblin-metrics/src/main/java/org/apache/gobblin/metrics/GobblinMetrics.java

```diff
@@ -367,6 +374,20 @@ public void startMetricReportingWithFileSuffix(State state, String metricsFileSu
       oldMetricsFileSuffix += "." + metricsFileSuffix;
     }
     metricsReportingProps.setProperty(ConfigurationKeys.METRICS_FILE_SUFFIX, oldMetricsFileSuffix);
+
+    if (!Strings.isNullOrEmpty(metricsFileSuffix)) {
```

Comment:
I think GobblinMetrics should be agnostic to the execution mode of the job, and shouldn't have to worry about extracting mapper/reducer num from the task id. This block can be moved inside MRJobLauncherClass and the configurations such as MAPPER_TASK_NUM_KEY can be set there and added into the state object.